intermediate_notebooks/E2E/mortgage/mortgage_e2e.ipynb

```diff
@@ -0,0 +1,899 @@
+    "    (arrow_df, list(client.who_has(arrow_df).values())[0][0])\n",
+    "    for arrow_df in arrow_dfs\n",
+    "]\n",
+    "new_map = OrderedDict()\n",
+    "for key, value in tmp_map:\n",
+    "    if value not in new_map:\n",
+    "        new_map[value] = [key]\n",
+    "    else:\n",
+    "        new_map[value].append(key)\n",
+    "\n",
+    "del (tmp_map, key, value)\n",
+    "\n",
+    "train_x_y = []\n",
+    "for list_delayed in new_map.values():\n",
+    "    train_x_y.append(delayed(prepare_data)(list_delayed))\n",
+    "\n",
+    "del (new_map, list_delayed)\n",
+    "\n",
+    "worker_list = OrderedDict()\n",
+    "for task in train_x_y:\n",
+    "    worker_list[task] = list(client.who_has(task).values())[0][0]\n",
+    "\n",
+    "del task\n",
+    "\n",
+    "persisted_train_x_y = []\n",
+    "for task in train_x_y:\n",
+    "    persisted_train_x_y.append(\n",
+    "        client.persist(\n",
+    "            collections=task,\n",
+    "            workers=worker_list[task],\n",
+    "            optimize_graph=False,\n",
+    "            fifo_timeout=\"0ms\",\n",
+    "        )\n",
+    "    )\n",
+    "\n",
+    "del (arrow_dfs, train_x_y, worker_list, task)\n",
+    "gc.collect()\n",
+    "\n",
+    "_ = wait(persisted_train_x_y)\n",
+    "del(_)\n",
+    "print(\"Prepared data for XGB training\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_x_y = dask_cudf.from_delayed(persisted_train_x_y)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "CPU times: user 604 ms, sys: 105 ms, total: 708 ms\n",
+      "Wall time: 4.27 s\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "dmat = xgb.dask.DaskDMatrix(client=client, \n",
+    "                     data=train_x_y[train_x_y.columns.difference([\"delinquency_12\"])],\n",
+    "                     label=train_x_y[[\"delinquency_12\"]],\n",
+    "                     missing=-1)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 24,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "458"
+      ]
+     },
+     "execution_count": 24,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "del(persisted_train_x_y, train_x_y)\n",
+    "gc.collect()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Train the Gradient Boosted Decision Tree with a single call to \n",
+    "```python\n",
+    "dask_xgboost.train(client, params, data, labels, num_boost_round=dxgb_gpu_params['nround'])\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "bst = xgb.dask.train(client,dxgb_gpu_params,dmat)"
```

Comment:
Seeing an increase in memory usage here.
cc: @JohnZed .

Quick question `xgb.Dask.DMatrix` causes the lazy computations to be materialized, but does it also create the matrix objects? If not, what happens to the persisted dataframe once the dmatrix object is created? Is it free from memory?